app/services/text_extraction.rb

```diff
@@ -44,15 +44,23 @@ def wf_name
 
   # mapping of cocina resource types to workflow names
   def resource_type_mapping
-    {
-      'https://cocina.sul.stanford.edu/models/book' => 'ocrWF',
-      'https://cocina.sul.stanford.edu/models/document' => 'ocrWF',
-      'https://cocina.sul.stanford.edu/models/image' => 'ocrWF'
-    }
+    {}.tap do |mapping|
+      if Settings.features.ocr_workflow
+        mapping['https://cocina.sul.stanford.edu/models/book'] = 'ocrWF'
+        mapping['https://cocina.sul.stanford.edu/models/document'] = 'ocrWF'
+        mapping['https://cocina.sul.stanford.edu/models/image'] = 'ocrWF'
+      end
+      mapping['https://cocina.sul.stanford.edu/models/media'] = 'speechToTextWF' if Settings.features.speech_to_text_workflow
+    end
```

Comment:
making the mapping dependent on the feature settings means the `.possible?` method in this class that is used to decide if we should render the "Text Extraction" button will work as expected based on the settings